ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -871,13 +991,110 @@ applyByronGenTx validationMode (ByronEBBLedgerConfig (ByronLedgerConfig cfg)) ge
 
-mkByronTx :: CC.UTxO.ATxAux ByteString -> GenTx (ByronBlockOrEBB cfg)
-mkByronTx tx = ByronTx
-    { byronTx   = tx
-    , byronTxId = Crypto.hash $ CC.UTxO.taTx tx
-      -- TODO replace this with a function from cardano-ledger, see
-      -- cardano-ledger#581
-    }
+    go (ByronDlg _ cert) cvs = wrapCVS <$>
+        V.Interface.updateDelegation env dlgState [cert]
+          `wrapError` ByronApplyDlgError
+      where
+        wrapCVS dlgState' = cvs { CC.Block.cvsDelegationState = dlgState' }
+
+        protocolMagic = Crypto.getAProtocolMagicId $ fixPM $
+          CC.Genesis.configProtocolMagic cfg
+
+        k = CC.Genesis.configK cfg
+
+        dlgState = CC.Block.cvsDelegationState cvs
+
+        currentEpoch =
+          CC.Slot.slotNumberEpoch (CC.Genesis.configEpochSlots cfg)
+                                  currentSlot
+
+        currentSlot = CC.Block.cvsLastSlot cvs
+
+        env = V.Interface.Environment
+          { V.Interface.protocolMagic     = protocolMagic
+          , V.Interface.allowedDelegators = allowedDelegators cfg
+          , V.Interface.k                 = k
+          , V.Interface.currentEpoch      = currentEpoch
+          , V.Interface.currentSlot       = currentSlot
+          }
+
+        fixPM (Crypto.AProtocolMagic a b) =
+          Crypto.AProtocolMagic (reAnnotate a) b
+
+    -- Note that we're only expecting update-related 'GenTx's in this case.
+    -- Both of them share much of the same code so it didn't seem to make
+    -- sense defining two separate cases of 'go'.
+    go tx cvs = wrapCVS <$> case tx of
+        ByronUpdateProposal _ proposal ->
+          CC.UPI.registerProposal env updateState proposal
+            `wrapError` ByronApplyUpdateProposalError
+
+        ByronUpdateVote _ vote ->
+          CC.UPI.registerVote env updateState vote
+            `wrapError` ByronApplyUpdateVoteError
+
+        _ -> error $  "applyByronGenTx.go: "
+                   <> "Only expecting update-related `GenTx`s at this point."
+      where
+        wrapCVS updateState' = cvs { CC.Block.cvsUpdateState = updateState' }
+
+        protocolMagic = Crypto.getAProtocolMagicId $ fixPM $
+          CC.Genesis.configProtocolMagic cfg
+
+        k = CC.Genesis.configK cfg
+
+        updateState = CC.Block.cvsUpdateState cvs
+
+        currentSlot = CC.Block.cvsLastSlot cvs
+
+        numGenKeys = toNumGenKeys $ Set.size (allowedDelegators cfg)
+
+        delegationMap =
+          (V.Interface.delegationMap . CC.Block.cvsDelegationState) cvs
+
+        env = CC.UPI.Environment
+          { CC.UPI.protocolMagic = protocolMagic
+          , CC.UPI.k             = k
+          , CC.UPI.currentSlot   = currentSlot
+          , CC.UPI.numGenKeys    = numGenKeys
+          , CC.UPI.delegationMap = delegationMap
+          }
+
+        fixPM (Crypto.AProtocolMagic a b) =
+          Crypto.AProtocolMagic (reAnnotate a) b
+
+        toNumGenKeys :: Integral n => n -> Word8
```

Comment:
I borrowed this from a module in ledger (it's not exported). It should probably be exported from ledger's update validation interface.